module/VuFind/src/VuFind/ILS/Driver/Folio.php

```diff
@@ -304,15 +326,38 @@ protected function renewTenantToken()
      */
     protected function checkTenantToken()
     {
-        $response = $this->makeRequest('GET', '/users', [], [], [401, 403]);
-        if ($response->getStatusCode() >= 400) {
+        if ($this->useLegacyAuthentication()) {
+            $response = $this->makeRequest('GET', '/users', [], [], [401, 403]);
+            if ($response->getStatusCode() >= 400) {
+                $this->token = null;
+                $this->tokenExpiration = null;
+                $this->renewTenantToken();
+                return false;
+            }
+            return true;
```

Comment:
Maybe it's better to be explicit, but this could perhaps be made less redundant like this:

```suggestion
            if ($response->getStatusCode() < 400) {
                return true;
            }
            // Clear token data to ensure that checkTenantTokenExpired triggers a renewal:
            $this->token = $this->tokenExpiration = null;
```

Reply:
I had that same debate too, less redundancy vs easier to read. Looking at the updated code it doesn't seem overly hard to follow so I've made that update.